test/unit/eigensolver/mc/test_reduction_to_band.cpp

```diff
@@ -0,0 +1,320 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include "dlaf/eigensolver/reduction_to_band.h"
+
+#include <cmath>
+
+#include <gtest/gtest.h>
+
+#include "dlaf/common/index2d.h"
+#include "dlaf/common/pipeline.h"
+#include "dlaf/communication/communicator.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/communication/functions_sync.h"
+#include "dlaf/communication/sync/broadcast.h"
+#include "dlaf/lapack/tile.h"  // Just for lapack.hh workaround
+#include "dlaf/matrix/copy.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/memory/memory_view.h"
+#include "dlaf/types.h"
+#include "dlaf/util_matrix.h"
+
+#include "dlaf_test/comm_grids/grids_6_ranks.h"
+#include "dlaf_test/matrix/matrix_local.h"
+#include "dlaf_test/matrix/util_matrix.h"
+#include "dlaf_test/matrix/util_tile.h"
+#include "dlaf_test/util_types.h"
+
+using namespace dlaf;
+using namespace dlaf::test;
+using namespace dlaf::comm;
+using namespace dlaf::matrix;
+using namespace dlaf::matrix::test;
+using namespace testing;
+
+::testing::Environment* const comm_grids_env =
+    ::testing::AddGlobalTestEnvironment(new dlaf::test::CommunicatorGrid6RanksEnvironment);
+
+template <typename Type>
+class ReductionToBandTest : public ::testing::Test {
+public:
+  const std::vector<CommunicatorGrid>& commGrids() {
+    return dlaf::test::comm_grids;
+  }
+};
+
+TYPED_TEST_SUITE(ReductionToBandTest, dlaf::test::MatrixElementTypes);
+
+const std::vector<LocalElementSize> square_sizes{{3, 3}, {13, 13}, {12, 12}, {24, 24}};
+const std::vector<TileElementSize> square_block_sizes{{3, 3}};
```

Comment:
can we try some more block sizes?

Reply:
do you have any suggestion?

Reply:
{2, 2}, {4, 4}?

Reply:
I would use a tuple {size, block_size} as in other tests to avoid exponential grow of test cases.